site/src/pages/__tests__/search.js

```diff
@@ -132,22 +151,34 @@ describe("Search", () => {
     )
   })
 
-  it("the query sanitizer is called with the query when updateSearchResults is called", () => {
+  it("the query sanitizer is called with the query when updateSearchResultsCallback is called", () => {
     const evt = { target: { value: "abc" } }
-    var index = new Index()
-    ;["title", "roles", "tags", "description", "body"].forEach(name =>
-      index.addField(name)
-    )
-    const data = {
-      allMarkdownRemark: articleList,
-      siteSearchIndex: { index: index.toJSON() }
-    }
+    
     expect(mockSanitizer.mock.calls.length).toEqual(0)
     const search = renderer.create(<Search data={data} debounceDelay={0} />)
-    search.getInstance().updateSearchResults(evt)
+    search.getInstance().updateSearchResultsCallback(evt)
 
     expect(mockSanitizer.mock.calls.length).toEqual(1)
     const sanitizerArguments = mockSanitizer.mock.calls[0]
     expect(sanitizerArguments[0]).toEqual("abc")
   })
+
+  it("stores the query on search bar update", async () => {
+    const query = "i am a test query"
+
+    const primarySearch = renderer.create(
+      <Search data={data} debounceDelay={0} />
+    )
+    primarySearch.getInstance().updateIndexedSearchValue(query)
+    expect(mockStore).toHaveBeenCalledWith(query)
+ })
+
+ it("when search is instantiated then retrieve is called", async () => {
+  const search = renderer.create(
```

Comment:
Check the retrieve function has not bee called before instantiating the widget.

Reply:
done